### PR TITLE
[FLINK-25525][FLINK-25487][table] Various fixed related to the new planner-loader

### PR DESCRIPTION
**Pull request by slinkydeveloper**
## What is the purpose of the change

This PR contains a couple of fixes and leftovers from the planner-loader PR.

## Brief change log

* Fix FLINK-25525
* Fix FLINK-25487
* Fix bad shading path for com.jayway

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-examples/flink-examples-table/pom.xml [hunk 1]
```diff
@@ -53,34 +53,42 @@ under the License.
...
 		<dependency>
 			<groupId>org.apache.flink</groupId>
-			<artifactId>flink-connector-files</artifactId>
+			<artifactId>flink-table-runtime</artifactId>
 			<version>${project.version}</version>
 		</dependency>
 		<dependency>
 			<groupId>org.apache.flink</groupId>
-			<artifactId>flink-csv</artifactId>
+			<artifactId>flink-table-planner_${scala.binary.version}</artifactId>
```
Comment (twalthr) [inline flink-examples/flink-examples-table/pom.xml:72]: Provided? Or is it somehow excluded from the JAR files we generate for the dist.

File: flink-examples/flink-examples-table/pom.xml [hunk 2]
```diff
@@ -53,34 +53,42 @@ under the License.
...
 
-		<!-- Table connectors and formats -->
+		<!-- The following two dependencies are not required to define a SQL job pipeline,
+		but only to execute it.
+
+		In particular, here we're forced to use flink-table-planner_${scala.binary.version} instead of
+		flink-table-planner-loader, because otherwise you hit this bug https://youtrack.jetbrains.com/issue/IDEA-93855
+		when trying to run the examples from within Intellij IDEA.
+
+		In a real environment, you need these dependencies only at test scope, to execute the SQL job pipelines test.
```
Comment (twalthr) [inline flink-examples/flink-examples-table/pom.xml:63]: `at test or provided scope (for tests or running it in an IDE).`

File: flink-examples/flink-examples-table/pom.xml [hunk 3]
```diff
@@ -53,34 +53,42 @@ under the License.
 			<version>${project.version}</version>
 		</dependency>
```
Comment (twalthr) [inline flink-examples/flink-examples-table/pom.xml:55]: remove empty line

File: flink-table/flink-table-planner/pom.xml [hunk 4]
```diff
@@ -196,14 +189,15 @@ under the License.
 			<!-- For legacy string expressions in Table API -->
 			<groupId>org.scala-lang.modules</groupId>
 			<artifactId>scala-parser-combinators_${scala.binary.version}</artifactId>
+			<scope>provided</scope>
```
Comment (twalthr) [inline flink-table/flink-table-planner/pom.xml:192]: Not sure if this is provided. It is not contained in Scala 2.12 by default.

File: flink-table/flink-table-planner/pom.xml [hunk 5]
```diff
@@ -196,14 +189,15 @@ under the License.
...
 		</dependency>
 
 		<dependency>
 			<!-- Utility to scan classpaths -->
 			<groupId>org.reflections</groupId>
 			<artifactId>reflections</artifactId>
 			<version>0.9.10</version>
-			<scope>compile</scope>
+			<scope>provided</scope>
```
Comment (twalthr) [inline flink-table/flink-table-planner/pom.xml:200]: Who pulls in this? not table-runtime.

**Review by twalthr — CHANGES_REQUESTED** [review]
Thanks @slinkydeveloper. I did a quick scan, will try it out locally in the next iteration.

File: flink-table/flink-table-planner/pom.xml (hunk 5 above)
Comment (slinkydeveloper) [reply flink-table/flink-table-planner/pom.xml:200]: Check `ReflectionsUtil`. We can get rid of this dep once during the FLIP-190 related work.

File: flink-table/flink-table-planner/pom.xml (hunk 4 above)
Comment (slinkydeveloper) [reply flink-table/flink-table-planner/pom.xml:192]: It doesn't matter, as we package it in table planner, so it doesn't need to be pulled in transitively for modules depending on `flink-table-planner` directly.

File: flink-examples/flink-examples-table/pom.xml (hunk 1 above)
Comment (slinkydeveloper) [reply flink-examples/flink-examples-table/pom.xml:72]: I think it makes no difference, as this pom doesn't produce a fat jar. In any case, I put it as provided more for documentation purpose.

**Comment by slinkydeveloper** [comment]
@twalthr I removed all provided, as I misunderstood their interaction with the shade plugin. please check again the first commit

**Review by twalthr — APPROVED** [review]
LGTM

**Comment by twalthr** [comment]
Merged via 137b65c5f23e5ba546ce52bd15fa7aa528ab95de

**Pull request closed without merging**